enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -366,9 +482,48 @@ type IstioCSRStatus struct {
 	// serviceAccount created by the controller for the istio-csr agent.
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 
+	// clusterRole created by the controller for the istio-csr agent.
+	ClusterRole string `json:"clusterRole,omitempty"`
+
 	// clusterRoleBinding created by the controller for the istio-csr agent.
 	ClusterRoleBinding string `json:"clusterRoleBinding,omitempty"`
 }
+
+type ConditionalStatus struct {
+	// conditions holds information of the current state of the istio-csr agent deployment.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ConfigMapReference holds the details of a configmap.
+type ConfigMapReference struct {
+	// name of the configmap.
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// namespace in which the configmap exists. If empty, configmap will be looked up in operator namespace.
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:Optional
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Cannot we restrict the presence of the ConfigMap only inside the namespace where `IstioCSR` resource will be created, or inside `IstioCSRConfig.Istio.Namespace`?

Currently, we are reading the secret present inside `IstioCSRConfig.Istio.Namespace` namespce, and creating the configmap inside `istiocsr.GetNamespace()` namespace. 

xref: https://github.com/openshift/cert-manager-operator/blob/baafb49bd6e645b6413a8b6b816816102ac6b7a6/pkg/controller/istiocsr/deployments.go#L363-L378

Reply:
Okay, so there could be 3 options?
- User configured namespace
- `istiocsr.GetNamespace()` e.g `istio-system` 
- `cert-manager-operator` namespace

Is my understanding correct?

Reply:
More context added on the ConfigMap usage.